### PR TITLE
Enable changing tracking rates

### PR DESCRIPTION
**Pull request by chemistorge**
Calculation of tracking rate (assuming a gear ratio of 101.25) was actually solar tracking rate (24 h) not sidereal.

Code has been change to include the ability to change  between solar, sidereal and lunar. Assuming Solar (24 h), sidereal (23 h 56 min) and lunar (24 h 31 min). 

The webpage code is updated. 

The code works with the current version of the app (the tracking rate defaults to sidereal if not set) but app would need update to include this functionality.

### CODE REVIEW

**Comment by Th0MmyS** [comment]
@chemistorge hi, thanks for the PR, I will check it out....does it make any difference during the night when you pick different tracking rate?

**Comment by chemistorge** [comment]
Depends on the focal length , the higher the focal length and longer subs the more important. It is also implementing a feature of more expensive tracking mounts.

File: esp32_wireless_control/firmware/strings.cpp [hunk 1]
```diff
@@ -4,6 +4,7 @@
 const char* TRACKING_ON = "Tracking ON";
 const char* TRACKING_OFF = "Tracking OFF";
 const char* SPEED = "speed";
+const char* TSPEED = "tspeed";
```
Comment (Th0MmyS) [inline esp32_wireless_control/firmware/strings.cpp:7]: please rename the constant to TRACKING_SPEED or anything thats more clear

File: esp32_wireless_control/firmware/firmware.ino [hunk 2]
```diff
@@ -23,22 +23,30 @@ const int dither_intensity = 5;
...
+                           TRACKING_SOLAR = 5333333,     //SOLAR (24h)
+                           TRACKING_LUNAR = 5447735 };   //LUNAR (24h, 31 min)
 const uint32_t c_SLEW_SPEED = SLEW_SPEED / 2;
 const int arcsec_per_step = 4;
 #endif
 
-int slew_speed = 0, exposure_count = 0, exposure_duration = 0, dither_enabled = 0, focal_length = 0, pixel_size = 0, steps_per_10pixels = 0, direction = c_DIRECTION;
+volatile enum tracking_rate_state tracking_rate = TRACKING_SIDEREAL; //default to sidereal rate
+
+int slew_speed = 0,
```
Comment (Th0MmyS) [inline esp32_wireless_control/firmware/firmware.ino:41]: you don't have to reformat this piece of code right? I mean the new line above slew_speed declaration and line break after slew_speed

File: esp32_wireless_control/firmware/firmware.ino [hunk 3]
```diff
@@ -140,6 +164,7 @@ void handleRight() {
   server.send(200, MIME_TYPE_TEXT, SLEWING);
 }
 
+
```
Comment (Th0MmyS) [inline esp32_wireless_control/firmware/firmware.ino:167]: why new line?

**Comment by chemistorge** [comment]
For example at 400 mm, a 4 minute sub at solar rate would have an error 10 arc seconds slower and lunar 86 arc sec. 

A take an example of the 90d with a 1.64 arc secs per pixel would mean a 6 pixel difference (solar vs sidereal) and 56 pixel difference with lunar (would not  need long subs).  With a small astro cam such as the ASI178 it would be 8 pixels. 

These are small errors but they all add up.

File: esp32_wireless_control/firmware/strings.cpp [hunk 4]
```diff
@@ -243,7 +259,7 @@ const char* html = "<!DOCTYPE html>\n"
                    "                </select><br>\n"
                    "            </div>\n"
                    "            <div class=\"button-group\">\n"
-                   "                <button class=\"right-separator\" type=\"button\" onclick=\"sendSlewRequest('/left')\">&#x2B9C;</button>\n"
+                   "                <button class=\"right-separator\" type=\"button\" onclick=\"sendSlewRequest('/left')\">&#11164;</button>\n"
```
Comment (Th0MmyS) [inline esp32_wireless_control/firmware/strings.cpp:262]: the original symbol was not working? it seems they are the same https://www.compart.com/en/unicode/U+2B9C

Comment (chemistorge) [reply esp32_wireless_control/firmware/strings.cpp:262]: I was trying to fix the symbol but could not fix it, tried different codes but no luck.

**Comment by Th0MmyS** [comment]
> For example at 400 mm, a 4 minute sub at solar rate would have an error 10 arc seconds slower and lunar 86 arc sec.
> 
> A take an example of the 90d with a 1.64 arc secs per pixel would mean a 6 pixel difference (solar vs sidereal) and 56 pixel difference with lunar (would not need long subs). With a small astro cam such as the ASI178 it would be 8 pixels.
> 
> These are small errors but they all add up.

I see....not really sure if you can reach these long subs on this type of tracker, even on V2....Ondra managed to take 2-3 minutes on 300mm on V2 ..... but I guess it will definitely help with the sky "walking away" from the frame

File: esp32_wireless_control/firmware/strings.cpp (hunk 4 above)
Comment (Th0MmyS) [reply esp32_wireless_control/firmware/strings.cpp:262]: if this symbol is not working, lets replace it with something more simple... or replace the symbol with actual words like "left" or "right

**Comment by Th0MmyS** [comment]
I prepared relevant change to Android app so when we'll fix the issues, I will release the Android app as well

**Comment by chemistorge** [comment]
Made changes and add new commit

**Comment by Th0MmyS** [comment]
Looks great :) thank you so much for the contribution... are you on discord server? I would like to mention you there

**Comment by chemistorge** [comment]
Yes I am on Discord

File: esp32_wireless_control/firmware/firmware.ino [hunk 5]
```diff
@@ -108,15 +114,31 @@ void handleRoot() {
 }
 
 void handleOn() {
+  int tracking_speed = server.arg(TRACKING_SPEED).toInt();
```
Comment (Th0MmyS) [inline esp32_wireless_control/firmware/firmware.ino:117]: just one more question.... when someone dont update their android app, will this be or when the app dont send the TRACKING_SPEED argument or will it crash the arduino?

Comment (chemistorge) [reply esp32_wireless_control/firmware/firmware.ino:117]: App does not crash, as I have add default to the switch case, which means is no valid tracking-speed it defaults to sidereal

Comment (Th0MmyS) [reply esp32_wireless_control/firmware/firmware.ino:117]: yeah I know you have the default branch there but just to be sure that the mapping toInt will not break

Comment (chemistorge) [reply esp32_wireless_control/firmware/firmware.ino:117]: tested with the current app version and it seems happy, it calls thr onHandle function and turns on tracking, and updates app.

Comment (Th0MmyS) [reply esp32_wireless_control/firmware/firmware.ino:117]: ok, thank you for testing, Im not able to test it myself right now....i'll merge your branch....thank you once again for your work :)

**Pull request merged**